### PR TITLE
Added cirq.state_vector_to_probabilities

### LINKED ISSUE #4322: Add `cirq.state_vector_to_probabilities`
**Is your feature request related to a use case or problem? Please describe.**
I need a function that takes in a `state_vector` and returns the probabilities of it's amplitudes.

**Describe the solution you'd like**
Add a `def state_vector_to_probabilities()` to 
https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/value/probability.py

**[optional] Describe alternatives/workarounds you've considered**
Defined my own:
```python
from cirq.value.probability import validate_probability
def state_vector_to_probabilities(state_vector: np.ndarray) -> np.ndarray:
      probabilities = np.abs(state_vector) ** 2
      validate_probability(probabilities, "sum of the amplitudes of the state vector")
      return probabilities

```
**What is the urgency from your perspective for this issue? Is it blocking important work?**
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #4322

### CODE REVIEW

File: cirq-core/cirq/value/probability.py [hunk 1]
```diff
@@ -33,3 +35,7 @@ def validate_probability(p: float, p_str: str) -> float:
     elif p > 1:
         raise ValueError(f'{p_str} was greater than 1.')
     return p
+def state_vector_to_probabilities(state_vector: np.ndarray) -> np.ndarray:
+      probabilities = np.abs(state_vector) ** 2
+      validate_probability(probabilities, "sum of the amplitudes of the state vector")
+      return probabilities
```
Comment (vtomole) [inline cirq-core/cirq/value/probability.py:41]: Add a newline.

File: cirq-core/cirq/value/probability.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from cirq.value.probability import validate_probability
```
Comment (vtomole) [inline cirq-core/cirq/value/probability.py:14]: Remove this line.

**Review by vtomole — CHANGES_REQUESTED** [review]
Thanks for opening this! I've left some feedback. Please some tests for this feature.

File: cirq-core/cirq/value/probability.py [hunk 3]
```diff
@@ -33,3 +35,7 @@ def validate_probability(p: float, p_str: str) -> float:
     elif p > 1:
         raise ValueError(f'{p_str} was greater than 1.')
     return p
+def state_vector_to_probabilities(state_vector: np.ndarray) -> np.ndarray:
```
Comment (balopat) [inline cirq-core/cirq/value/probability.py:38]: I think we should be using `cirq.to_valid_state_vector` to convert the argument (which should be of type `'cirq.STATE_VECTOR_LIKE'`)  to a valid state vector. 
After that there is no need for `validate_probability` either.

**Review by balopat — CHANGES_REQUESTED** [review]

**Comment by vtomole** [comment]
@Patil2099 Ping

File: cirq-core/cirq/value/probability.py (hunk 3 above)
Comment (vtomole) [reply cirq-core/cirq/value/probability.py:38]: Agree. This is being done on https://github.com/quantumlib/Cirq/pull/4439.

**Comment by vtomole** [comment]
Duplicate of https://github.com/quantumlib/Cirq/pull/4439

**Pull request closed without merging**